src/txgraph.cpp

```diff
@@ -1311,13 +1311,13 @@ void SingletonClusterImpl::AppendChunkFeerates(std::vector<FeeFrac>& ret) const
 
 uint64_t GenericClusterImpl::AppendTrimData(std::vector<TrimTxData>& ret, std::vector<std::pair<GraphIndex, GraphIndex>>& deps) const noexcept
 {
-    const LinearizationChunking linchunking(m_depgraph, m_linearization);
+    auto linchunking = ChunkLinearizationInfo(m_depgraph, m_linearization);
     LinearizationIndex pos{0};
     uint64_t size{0};
     auto prev_index = GraphIndex(-1);
     // Iterate over the chunks of this cluster's linearization.
-    for (unsigned i = 0; i < linchunking.NumChunksLeft(); ++i) {
-        const auto& [chunk, chunk_feerate] = linchunking.GetChunk(i);
+    for (unsigned i = 0; i < linchunking.size(); ++i) {
```

Comment:
26661a6c655ad148d3ecf200c64b89c439b6d44a

nit: while we're here (we also arent using `j` below anywhere, but meh)

```Suggestion
    for (const auto& [chunk, chunk_feerate] : linchunking) {
```